backend/ethereum/channel/adjudicator_test.go

```diff
@@ -33,6 +34,128 @@ import (
 	pkgtest "perun.network/go-perun/pkg/test"
 )
 
+// TestAdjudicator tests the adjudicator.
+func TestAdjudicator(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	rng := pkgtest.Prng(t)
+	numParts := 2 + rng.Intn(maxNumParts-2)
+	s := test.NewSetup(t, rng, numParts, blockInterval)
+	a := newAdjudicatorSetup(s)
+	channeltest.TestAdjudicator(ctx, t, a)
+}
+
+type adjudicatorSetup struct {
+	setup *test.Setup
+}
+
+func newAdjudicatorSetup(setup *test.Setup) *adjudicatorSetup {
+	return &adjudicatorSetup{
+		setup: setup,
+	}
+}
+
+type testAdjudicator struct {
+	*test.SimAdjudicator
+}
+
+// Progress is a no-op because app channels are not supported yet.
+func (a *testAdjudicator) Progress(ctx context.Context, req channel.ProgressReq) error {
+	return nil
+}
+
+func (a *adjudicatorSetup) Adjudicator() channel.Adjudicator {
+	return &testAdjudicator{a.setup.Adjs[0]}
+}
+
+func (a *adjudicatorSetup) NewRegisterReq(ctx context.Context, rng *rand.Rand) (*channel.AdjudicatorReq, []channel.SignedState) {
+	req := a.newAdjudicatorReq(ctx, rng, channeltest.WithIsFinal(false))
+	return req, nil
+}
+
+func (a *adjudicatorSetup) newAdjudicatorReq(ctx context.Context, rng *rand.Rand, opts ...channeltest.RandomOpt) *channel.AdjudicatorReq {
+	params, state := a.newRandomParamsAndState(rng, opts...)
+
+	// Fund the channel.
```

Comment:
I think we could limit the generic test to only make calls on already funded channels.  
This could be done by embedding the `Funder interface` into the `Adjudictor interface` and then let *go-perun* fund the channels before using them.

Reply:
Yeah, but wouldn't it be somewhat unclean to embed the funder into the adjudicator tests? Maybe for some backends there is another, easier way to set things up. I am not sure, I tried to keep the interfaces and test code simple and minimal at the start, let's discuss in a call.